src/Illuminate/Database/Connection.php

```diff
@@ -845,6 +846,17 @@ protected function getElapsedTime($start)
         return round((microtime(true) - $start) * 1000, 2);
     }
 
+    /**
+     * Get the affected row count.
+     *
+     * @param  mixed  $result
+     * @return int|null
+     */
+    protected function getRowCount($result)
+    {
+        return is_array($result) ? count($result) : (is_int($result) ? $result : null);
```

Comment:
wouldn't the default be `0` here instead of `null`?

Reply:
I think that if a count is not obtained it should be null, this way you could differentiate when it is actually zero.